kartothek/io/dask/dataframe.py

```diff
@@ -301,3 +303,47 @@ def update_dataset_from_ddf(
         metadata=metadata,
         metadata_merger=metadata_merger,
     )
+
+
+def collect_dataset_statistics(
+    store, dataset_uuid, table_name, predicates=None, frac=0.3
+):
+    """
+    Retrieve some statistics, maybe even plot them, generate a report...?
+
+    Parameters
+    ----------
+    store
+      A factory function providing a KeyValueStore
+    dataset_uuid
+      The dataset's unique identifier
+    table_name
+      Name of the kartothek table for which to retrieve the statistics
+    predicates
+      Kartothek predicates to apply filters on the data for which to gather statistics
+    frac
+      Fraction of the data to use for gathering statistics. `frac == 1` will use the whole dataset.
+
+    Returns
+    -------
+    A dask.dataframe containing information about dataset statistics.
+
+    """
+
+    def _get_parquet_stats(mp, store=store, table_name=table_name):
```

Comment:
Please use global scope to define the function, i.e. outside of the function context. This is more efficient for serialization

Reply:
Removed it entirely in favor of your proposed solution.